src/explorerCommands/new.ts

```diff
@@ -0,0 +1,108 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+
+import * as fse from "fs-extra";
+import * as path from "path";
+import { Uri, window, workspace, WorkspaceEdit } from "vscode";
+import { NodeKind } from "../java/nodeData";
+import { isJavaIdentifier, isKeyword } from "../utility";
+import { DataNode } from "../views/dataNode";
+
+export async function newJavaClass(node: DataNode): Promise<void> {
+    const packageFsPath: string = Uri.parse(node.uri).fsPath;
+    const className: string | undefined = await window.showInputBox({
+        placeHolder: "Input the class name",
+        ignoreFocusOut: true,
+        validateInput: async (value: string): Promise<string> => {
+            const checkMessage: string = checkJavaQualifiedName(value);
+            if (checkMessage) {
+                return checkMessage;
+            }
+
+            if (await fse.pathExists(getNewFilePath(packageFsPath, value))) {
+                return "Class already exists.";
+            }
+
+            return "";
+        },
+    });
+
+    if (!className) {
+        return;
+    }
+
+    // `workspace.applyEdit()` will trigger a workspace file event, and let the
+    // vscode-java extension to handle the type: class, interface or enum.
+    const workspaceEdit: WorkspaceEdit = new WorkspaceEdit();
+    const fsPath: string = getNewFilePath(packageFsPath, className);
+    workspaceEdit.createFile(Uri.file(fsPath));
+    workspace.applyEdit(workspaceEdit);
+}
+
+function getNewFilePath(basePath: string, className: string): string {
+    const indexOfJavaExt: number = className.lastIndexOf(".java");
```

Comment:
One corner case is "org.java.MyClass", looks like you only need check whether className.endsWith(".java")

Reply:
yes, make sense